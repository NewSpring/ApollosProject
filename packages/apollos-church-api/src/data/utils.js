import ApollosConfig from '@apollosproject/config';

const { ROCK, ROCK_MAPPINGS } = ApollosConfig;

export const createAssetUrl = (value) =>
  value.path
    ? value.path
    : `${ROCK.IMAGE_URL}/${
        ROCK_MAPPINGS.ASSET_STORAGE_PROVIDERS[`${value.AssetStorageProviderId}`]
      }/${value.Key}`;

export const fakeFunction = () => 'Fake';
