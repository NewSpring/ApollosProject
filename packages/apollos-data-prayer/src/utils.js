import ApollosConfig from '@apollosproject/config';
import { AuthenticationError } from 'apollo-server';
import jwt from 'jsonwebtoken';

export const secret = process.env.SECRET || 'ASea$2gadj#asd0';

export const parseToken = (token) => jwt.verify(token, secret);

export const registerToken = (token) => {
  try {
    const { cookie, sessionId } = parseToken(token);

    return {
      userToken: token,
      rockCookie: cookie,
      sessionId,
    };
  } catch (e) {
    throw new AuthenticationError('Invalid token');
  }
};

export const generateToken = (params) =>
  jwt.sign({ ...params }, secret, { expiresIn: '60d' });

export const enforceProtocol = (uri) =>
  uri.startsWith('//') ? `https:${uri}` : uri;

export const createImageUrlFromGuid = (uri) =>
  uri.split('-').length === 5
    ? `${ApollosConfig.ROCK.IMAGE_URL}?guid=${uri}`
    : enforceProtocol(uri);

export const latLonDistance = (lat1, lon1, lat2, lon2) => {
  if (lat1 === lat2 && lon1 === lon2) {
    return 0;
  }
  const radlat1 = (Math.PI * lat1) / 180;
  const radlat2 = (Math.PI * lat2) / 180;
  const theta = lon1 - lon2;
  const radtheta = (Math.PI * theta) / 180;
  let dist =
    Math.sin(radlat1) * Math.sin(radlat2) +
    Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
  if (dist > 1) {
    dist = 1;
  }
  dist = Math.acos(dist);
  dist = (dist * 180) / Math.PI;
  dist = dist * 60 * 1.1515;
  return dist;
};
