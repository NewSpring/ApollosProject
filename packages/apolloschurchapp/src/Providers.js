import React from 'react';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

import theme from './theme'; // where theme looks like:

const NewSpringProviders = (props) => (
  <Providers themeInput={theme}>
    <ClientProvider {...props} />
  </Providers>
);

export default NewSpringProviders;
