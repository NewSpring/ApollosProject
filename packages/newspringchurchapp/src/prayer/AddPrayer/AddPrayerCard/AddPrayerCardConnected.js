import React, { memo } from 'react';
import { Query } from 'react-apollo';
import { get } from 'lodash';
import getProfilePic from '../../getProfilePic';
import AddPrayerCard from './AddPrayerCard';

const AddPrayerCardConnected = memo(({ ...props }) => (
  <Query query={getProfilePic} fetchPolicy={'cache-only'}>
    {({ data }) => (
      <AddPrayerCard
        imgSrc={get(data, 'currentUser.profile.photo', { uri: '' })}
        {...props}
      />
    )}
  </Query>
));

AddPrayerCardConnected.displayName = 'AddPrayerCardConnected';

export default AddPrayerCardConnected;
