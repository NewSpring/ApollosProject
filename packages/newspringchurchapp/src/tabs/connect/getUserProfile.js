import gql from 'graphql-tag';
import CampusParts from 'newspringchurchapp/src/user-settings/Locations/campusFragment';

export default gql`
  query getCurrentUserProfile {
    currentUser {
      id
      profile {
        id
        impersonationParameter
        firstName
        lastName
        campus {
          ...CampusParts
        }
        email
        nickName
        gender
        birthDate
        photo {
          uri
        }
      }
    }
  }
  ${CampusParts}
`;
