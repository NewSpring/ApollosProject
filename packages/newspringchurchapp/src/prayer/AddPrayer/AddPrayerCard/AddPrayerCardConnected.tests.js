import React from 'react';

import Providers from 'newspringchurchapp/src/Providers';
import { renderWithApolloData } from 'newspringchurchapp/src/utils/testUtils';
import getUserProfile from 'newspringchurchapp/src/tabs/connect/getUserProfile';
import AddPrayerCardConnected from './AddPrayerCardConnected';

const mocks = [
  {
    request: {
      query: getUserProfile,
    },
    result: {
      data: {
        currentUser: { profile: { photo: { uri: 'fakeURL' } } },
      },
    },
  },
];

it('renders AddPrayerCardConnected', async () => {
  const tree = await renderWithApolloData(
    <Providers mocks={mocks}>
      <AddPrayerCardConnected />
    </Providers>
  );
  expect(tree).toMatchSnapshot();
});
