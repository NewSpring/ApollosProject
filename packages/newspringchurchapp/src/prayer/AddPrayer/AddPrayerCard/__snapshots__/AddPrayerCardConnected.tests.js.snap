// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddPrayerCardConnected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomLeftRadius": 16,
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 64,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "stretch",
              "flexDirection": "row",
              "flexWrap": "nowrap",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
              }
            }
          >
            Ask for prayer
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              bordered={false}
              disabled={false}
              pill={false}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#dddddd",
                  "borderColor": "#dddddd",
                  "borderRadius": 16,
                  "borderWidth": 2,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 32,
                  "justifyContent": "center",
                  "marginBottom": 8,
                  "marginRight": 8,
                  "opacity": 1,
                  "overflow": "hidden",
                  "paddingHorizontal": 4,
                  "paddingVertical": 4,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignItems": "center",
                    "color": "#303030",
                    "fontFamily": "Colfax-Bold",
                    "fontSize": 12,
                    "justifyContent": "center",
                    "lineHeight": 17.28,
                    "paddingHorizontal": 6,
                    "textAlign": "center",
                  }
                }
              >
                My Prayers
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#303030",
              "fontFamily": "System",
              "fontSize": 14,
              "fontStyle": null,
              "fontWeight": null,
              "lineHeight": 20.16,
              "paddingVertical": 16,
            }
          }
        >
          There aren't currently any prayers in your community. Send them one now.
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "padding": 16,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            bordered={false}
            disabled={false}
            pill={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#6bac43",
                "borderColor": "#6bac43",
                "borderRadius": 48,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Colfax-Bold",
                  "fontSize": 16,
                  "lineHeight": 23.04,
                }
              }
            >
              Add a Prayer Request
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
